jbmc/src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -32,9 +32,11 @@ Author: Daniel Kroening, kroening@kroening.com
 class java_bytecode_parsert:public parsert
 {
 public:
-  java_bytecode_parsert()
+  explicit java_bytecode_parsert(bool skip_instructions)
+    : skip_instructions(skip_instructions)
   {
-    get_bytecodes();
+    if(!skip_instructions)
+      get_bytecodes();
```

Comment:
I'm a bit surprised that we both need a class member and this branch here. But that may just be down to my limited understanding of the code here.

Reply:
The method is badly named: this actually initialises a lookup table of possible bytecodes (only used within rbytecodes)

Reply:
Maybe the `if(!skip_instructions)` should be moved into this method as `if(skip_instructions) return;` - you need to know what the method does in order to know whether you can skip it, and therefore to me it would seem better to place the condition with the implementation.